build/Microsoft.DotNet.Cli.Prepare.targets

```diff
@@ -164,24 +164,32 @@
                     InputFile="%(_DownloadAndExtractItem.DownloadFileName)"
                     DestinationDirectory="%(_DownloadAndExtractItem.ExtractDestintation)" />
   </Target>
-
-  <Target Name="SetupRestorePackagesInputsOutputs">
-    <ItemGroup>
-      <RestorePackagesInput Include="$(RepoRoot)/src/**/project.json" />
-      <RestorePackagesInput Include="$(RepoRoot)/tools/**/project.json" />
-    </ItemGroup>
-  </Target>
   
   <Target Name="RestorePackages"
           DependsOnTargets="SetupRestorePackagesInputsOutputs;Init"
           Inputs="@(RestorePackagesInput)"
           Outputs="@(RestorePackagesInput->'%(RelativeDir)/project.lock.json')">
 
+    <CallTarget Targets="CleanSrcToolsLockFiles" />
```

Comment:
Hmmm... I can see the reasoning. 
I guess this is giving me a smell because the CleanSrcToolsLockFiles is seemingly treated like a function here. I don't think that would be a good conceptual pattern for us to start down because targets can't run more than once, and I could see someone copy/pasting this line into their target without thinking somewhere down the line. 

I don't feel too strongly here though since the issue is mostly abstract.